capstone/src/main/webapp/script.js

```diff
@@ -162,6 +163,65 @@ function setUpReplacements(text, lang, substringForReplacements) {
   showDataMuseWordReplacementOptions(text, lang, substringForReplacements);
 }
 
+/** Gets the most extreme word replacements on a string */
+async function getExtremes(text, lang) {
```

Comment:
Is it correct that `text` here will not have changed since you last fetched DataMuse and perspective results?  If so, then maybe you can change the logic to save the results from the previous calls to DataMuse+Perspective and just find the extreme values in those, that way there aren't extra API calls?

Reply:
Yes, `text` will not have changed. However, I would usually only be able to store 5 Datamuse+Perspective API calls from the previous outputted results. These are sorted, so its very easy to see the most extreme out of those. This function was meant to be a cool way for the user to see what Perspective thinks is very toxic or not, and to catch outliers. To do that, I need to find a large number of sentence variations (I search for 15 replacements for every single word in `text`). This is a slow process and I don't see a way to optimize significantly (I could store those 5 results) without sacrificing accuracy, but there are sometimes interesting results.